File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    if (s.length%2!==0) return 0
    let tmp;
    const stack = []
    for(const i of s){
        if(tmp===i){
            stack.pop()
            tmp = stack.at(-1)
            continue
        }
        
        stack.push(i);
        tmp = i;
    }
    
    if(stack.length>0) return 0
    return 1
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        String[] strArr = s.split("");
        Deque<String> deque = new ArrayDeque<>();
        for(String str : strArr) {
            if(!deque.isEmpty() && str.equals(deque.peek())){
                deque.pop();
                continue;
            }
            deque.push(str);
        }
        
        return deque.size() == 0 ? 1 : 0;
    }
}